CIScripts/Test/Tests/vRouterAgentMSIInstaller.Tests.ps1

```diff
@@ -1,37 +1,55 @@
 Param (
-    [Parameter(Mandatory=$true)] [string] $TestbedAddr
+    [Parameter(Mandatory=$true)] [string] $TestbedAddr,
+    [Parameter(Mandatory=$true)] [string] $ConfigFile
 )
 
 . $PSScriptRoot\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\Utils\ComponentsInstallation.ps1
 . $PSScriptRoot\..\TestConfigurationUtils.ps1
 . $PSScriptRoot\..\..\Common\Aliases.ps1
 . $PSScriptRoot\..\..\Common\VMUtils.ps1
 . $PSScriptRoot\..\PesterHelpers\PesterHelpers.ps1
 
-. $PSScriptRoot\..\GetTestConfigurationCodiLegacy.ps1
+. $ConfigFile
 $TestConf = Get-TestConfiguration
 $Session = New-PSSession -ComputerName $TestbedAddr -Credential (Get-VMCreds)
 
 Describe "vRouter Agent MSI installer" {
-    Context "installation and uninstallation" {
-        It "registers/unregisters agent service and never enables" {
-            Install-Agent -Session $Session
-            Eventually {
-                Get-AgentServiceStatus -Session $Session | Should Be "Stopped"
-            } -Duration 3
-
-            Uninstall-Agent -Session $Session
-            Eventually {
-                {Get-AgentServiceStatus -Session $Session} | Should Throw
-            } -Duration 3
+
+    function Test-AgentMSIBehaviourCorrect {
+        Install-Agent -Session $Session
+        Eventually {
+            Get-AgentServiceStatus -Session $Session | Should Be "Stopped"
+        } -Duration 3
+
+        Uninstall-Agent -Session $Session
+        Eventually {
+            Get-AgentServiceStatus -Session $Session | Should Throw
```

Comment:
1. `Should Throw` always needs `{}`
2. You've refactored `Get-AgentServiceStatus` never to throw so you wanted `Should Be Null` probably.